client/src/modules/reports/generate/account_report_multiple/account_report_multiple.config.js

```diff
@@ -0,0 +1,107 @@
+angular.module('bhima.controllers')
+  .controller('account_report_multipleController', AccountReportMultipleConfigController);
+
+AccountReportMultipleConfigController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'reportData',
+  '$state', 'moment', 'SessionService',
+];
+
+function AccountReportMultipleConfigController(
+  $sce, Notify, SavedReports, AppCache, reportData, $state,
+  Moment, Session
+) {
+  const vm = this;
+  const cache = new AppCache('configure_account_report_multiple');
+  const reportUrl = 'reports/finance/account_report_multiple';
+  vm.previewGenerated = false;
+  vm.accountIds = [];
+  delete cache.reportDetails;
+  vm.reportDetails = {
+    currency_id : Session.enterprise.currency_id,
+    selectedAccounts : [],
+  };
+
+  vm.dateInterval = 1;
+
+  checkCachedConfiguration();
+
+  vm.selectAccount = function selectAccount(account) {
+    vm.reportDetails.selectedAccounts.push({ type_id : account.type_id, id : account.id });
+  };
+  vm.clearPreview = function clearPreview() {
+    vm.previewGenerated = false;
+    vm.previewResult = null;
+  };
+
+  vm.setCurrency = function setCurrency(currencyId) {
+    vm.reportDetails.currency_id = currencyId;
+  };
+
+  vm.onChangeAccounts = function onChangeAccounts(accountId) {
+    vm.reportDetails.selectedAccounts = vm.reportDetails.selectedAccounts.filter(ac => {
+      return ac.id !== accountId;
+    });
+  };
+
+  vm.requestSaveAs = function requestSaveAs() {
+    parseDateInterval(vm.reportDetails);
+
+    const options = {
+      url : reportUrl,
+      report : reportData,
+      reportOptions : sanitiseDateStrings(vm.reportDetails),
+    };
+
+    return SavedReports.saveAsModal(options)
+      .then(() => {
+        $state.go('reportsBase.reportsArchive', { key : options.report.report_key });
+      })
+      .catch(Notify.handleError);
+  };
+
+  vm.preview = function preview(form) {
+    if (form.$invalid) {
+      Notify.danger('FORM.ERRORS.RECORD_ERROR');
+      return 0;
+    }
+
+    parseDateInterval(vm.reportDetails);
+
+    vm.reportDetails.selectedAccounts = vm.reportDetails.selectedAccounts.filter(ac => {
+      return vm.accountIds.indexOf(ac.id);
+    });
```

Comment:
What is this statement doing?  It isn't clear to me.

Reply:
Here we are using a select multiple; when a selected account is removed , this snipet of code will  filter  ```vm.reportDetails.selectedAccounts```  then it will be reduced.
You can see at the screen shot bellow how accounts are displayed

![selected accounsts](https://user-images.githubusercontent.com/25838121/45346680-c38d9c80-b5a1-11e8-9ac4-0f506b74b3b6.PNG)

Reply:
You can notice that I need in the ui-select both then label and the account number but when the user remove the selected account, the on-remove callback return just the id of the account